client/src/app/project-registration/contact-field-group/contact-field-group.component.ts

```diff
@@ -40,30 +40,35 @@ export class ContactFieldGroupComponent implements OnInit {
   ngOnInit(): void {
     this.contributorsControl = this.metadataForm.getControl(this.contactKey);
     this.contributorMetadata = this.metadataForm.get(this.contactKey);
-    this.addFormControl(this.contributorMetadata, this.contributorsControl);
 
     const fieldList = this.contactFieldList;
-
     this.contactFieldMetadataList = fieldList.map(field => this.metadataForm.get(field));
 
-    const contactEmailCtrl = this.contributorsControl['controls'][0]['controls']['email'];
-    const contactNameCtrl = this.contributorsControl['controls'][0]['controls']['name'];
-    const correspondingCtrl = this.contributorsControl['controls'][0]['controls']['corresponding_contributor'];
-
-    // default
-    correspondingCtrl.setValue(true);
-
-    this.aai.getUser().subscribe(user => {
-      if (AaiService.loggedIn(user) && !this.userInfo && user.profile) {
-        this.userInfo = user.profile;
-        const previousValue = contactNameCtrl.value;
-        const name = [this.userInfo.given_name, '', this.userInfo.family_name].join(',');
-        if (previousValue !== name) {
-          contactNameCtrl.setValue(name);
-          contactEmailCtrl.setValue(this.userInfo.email);
+    // Default to having one contributor form item added, with the values filled by login user details,
+    // if no contributors have been added via the autofill service
+    if (!this.contributorsControl.value || !this.contributorsControl.value.length) {
+
+      this.addFormControl(this.contributorMetadata, this.contributorsControl);
+
+      const contactEmailCtrl = this.contributorsControl['controls'][0]['controls']['email'];
```

Comment:
Can you put `this.contributorsControl['controls'][0]['controls']` in another variable and reference that here? I think you could probably use object destructuring here to make it more readable as well

Reply:
I'm reading about object destructuring and it looks pretty cool. Thanks for telling me about it!

Reply:
leaving this link here in case anyone else wants to read about it: https://dmitripavlutin.com/javascript-object-destructuring/#:~:text=The%20object%20destructuring%20is%20a,the%20property%20doesn't%20exist.

I found that this image summarized it pretty well: https://dmitripavlutin.com/static/c6d9d0af2d1b6e8d0c88dad3f41f55f4/7842b/javascript-object-destructuring-infographic-2.png